### PR TITLE
Add example app for notifications

### PR DESCRIPTION
**Pull request by jonyoder**
* Adds an example application that demonstrates notifications.
* Corrects some help strings for existing examples.

## Usage

You must have a postgres server running and available to try the `pgx` and `pq` drivers.

### Local driver

To test with the `local` driver, which can only notify within a single process, use a command like this:
```sh
./out/testnotify notify --driver=local
```

### Postgres drivers

To test with the `pgx` and `pq` drivers, use commands like this. Try starting multiple commands in separate terminals and notice how all notifications go to all processes. You can even mix `pq` and `pgx` drivers.

```sh
./out/testnotify notify --driver=pgx
./out/testnotify notify --driver=pq
```

### Filtering

To test filtering, start filtering with one process, and then send notifications with another:

```sh
# Start filtering for the string "pq". This process will exit as soon as it receives a message
# that matches the regex "DONE".
./out/testnotify notify --driver=pgx --filter="DONE"

# Send a notification to match the filter.
./out/testnotify notify --driver=pq --message="DONE"
```

### Connection string

The pgx/pq drivers assume your postgres DB is available at `postgres` with admin/password as the user/password. You can pass a custom connection string like this:

```sh
./out/testnotify notify --driver=pgx --conn-str=postgres://admin:password@localhost/postgres?sslmode=disable
```

### CODE REVIEW

File: examples/cmd/testnotify/cmd/notify.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+				log.Printf("Received notification: %s", tn.Message)
+			}
+		}
+	}
+}
+
+// localNotify sends one notification per second with the provided local ListenerProvider.
+func localNotify(channelName string, lp *local.ListenerProvider, message *testNotification) {
+	tick := time.NewTicker(time.Second)
+	defer tick.Stop()
```
Comment (kfeinauer) [inline examples/cmd/testnotify/cmd/notify.go:105]: Does this actually execute?

File: examples/cmd/testapp/cmd/log.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ var (
 )
 
 func init() {
-	LogCmd.Example = `  rspm log --message=hello --level=INFO
+	LogCmd.Example = `  testapp log --message=hello --level=INFO
```
[the remark this answers is not in the document]
Comment (jonyoder) [reply examples/cmd/testapp/cmd/log.go:15]: For some reason, we've always written the help strings for Cobra commands this way. It's a somewhat common pattern in Go.

[the remark this answers is not in the document]
Comment (jonyoder) [reply examples/cmd/testapp/cmd/log.go:15]: And, yes, you're correct. In Go, strings surrounded by backticks include the newlines.

File: examples/cmd/testnotify/cmd/notify.go (hunk 1 above)
Comment (jonyoder) [reply examples/cmd/testnotify/cmd/notify.go:105]: It's a good practice whether it ends up executing or not. It's easy to leak tickers and end up with thousands of leaked go routines if you don't stop them.

File: examples/cmd/testnotify/cmd/notify.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+)
+
+func init() {
+	debugLogger = debug.NewDebugLogger(RegionNotify)
+
+	NotifyCmd.Example = `  testnotify notify
+  testnotify notify --driver=pq
+  testnotify notify --message="hello from pq" --driver=pq --conn-str=postgres://admin:password@localhost/postgres?sslmode=disable
+`
+	NotifyCmd.Flags().StringVar(&message, "message", "HELO from <driver>", "The message to send.")
```
[the remark this answers is not in the document]
Comment (jonyoder) [reply examples/cmd/testnotify/cmd/notify.go:43]: Haha. This was a silly play on the old SMTP `HELO` command. I actually forgot it was from SMTP and was thinking it was more common. 🤣

**Review by kfeinauer — APPROVED** [review]

**Pull request merged**